packages/gitbook/src/components/TableOfContents/TOCScroller.tsx

```diff
@@ -0,0 +1,78 @@
+'use client';
+
+import React from 'react';
+
+import { ClassValue, tcls } from '@/lib/tailwind';
+
+import { useHash } from '../hooks';
+
+const TOCScrollContainerContext = React.createContext<React.RefObject<HTMLDivElement> | null>(null);
+
+export function TOCScrollContainerProvider(props: {
+    children: React.ReactNode;
+    className?: ClassValue;
+}) {
+    const { children, className } = props;
+    const scrollContainerRef = React.createRef<HTMLDivElement>();
+
+    return (
+        <TOCScrollContainerContext.Provider value={scrollContainerRef}>
+            <div ref={scrollContainerRef} className={tcls(className)}>
```

Comment:
```suggestion
            <div ref={scrollContainerRef} className={className}>
```

Reply:
This project uses `tcls` and `twMerge` and this change results in a typescript error, ClassValue can't be assigned to `string | undefined`.